src/FizzWare.NBuilder/PropertyNaming/PropertyNamer.cs

```diff
@@ -88,15 +88,18 @@ private static bool IsNullableType(Type type)
 
         protected virtual void SetValue<T>(MemberInfo memberInfo, T obj, object value)
         {
-            if (value != null)
+            if (value == null) return;
+            if (memberInfo.Name == "Empty" && memberInfo.DeclaringType == typeof(Guid)) return;

```

Comment:
I don't think this is the right approach. The code shouldn't special-case Guid. The root problem is that NBuilder is altering a static readonly member field. The code should be modified to _not_ set members that are readonly and/or static.

Reply:
I started to think this too... This has been changed